src/utils/GpioInitializer.hxx

```diff
@@ -61,22 +61,22 @@ template <typename Head, typename... Tail>
 struct GpioInitHelper<std::tuple<Head, Tail...>>
 {
     /// Middle step of the recursion for an individual pin.
-    static void hw_init()
+    static void __attribute__((always_inline)) hw_init()
     {
         Head::hw_init();
-        GpioInitHelper<std::tuple<Tail...>>::hw_init();
+        return GpioInitHelper<std::tuple<Tail...>>::hw_init();
```

Comment:
I'm a bit confused why there is a return value in a void return method. Can you explain?

Reply:
You can return with a void function call from a void function. This is just a strong signal to the compiler that tail recursion optimization should be employed.